docs/content/3.middleware/2.guides/10.logging.md

```diff
@@ -0,0 +1,340 @@
+
+const testingExtension = {
+  name: "some-extension",
+  extendApiMethods: {
+    addedCustomEndpoint(context) {
+      const logger = getLogger(context);
+      logger.info("You can log inside newly created handler!");
+      return {
+        // ...
+      };
+    }
+  }
+};
+```
+
+:::tip
+You can derive logger from context the same way if you are creating an integration.
+:::
+
+### onCreate
+
+The hook got a new parameter from which you can derive the logger with help of `getLogger` function.
+
+```ts
+const onCreate = async (
+  config: Record<string, unknown> = {},
+  alokai: AlokaiContainer
+) => {
+  const logger = getLogger(alokai);
+  logger.info("You can log inside onCreate when creating a new integration");
+
+  return {
+    // ...
+  };
+};
+```
+
+### init
+
+The hook got a new parameter from which you can derive the logger with help of `getLogger` function.
+
+```ts
+const init = (settings: any, alokai: AlokaiContainer) => {
+  const logger = getLogger(alokai);
+  logger.info("You can log inside init function when creating an integration");
+
+  return {
+    config: settings,
+    client: null,
+  };
+};
+```
+
+## Configuration
+
+The middleware logger offers flexible configuration options to control logging behavior across the application. These options allow you to manage the verbosity and level of detail in logs based on the needs of different environments, such as development or production.
+
+### Available Configuration Options
+
+```ts
+/**
+ * Options for the logger.
+ */
+export interface LoggerOptions {
+  /**
+   * The log level aligned with RFC5424.
+   */
+  level?: LogLevel; // (def. "info")
+
+  /**
+   * Whether to include the stack trace in the log message.
+   */
+  includeStackTrace?: boolean; // (def. true)
+
+  /**
+   * Own implementation of logger to be used internally.
+   *
+   * @remarks If provided then other options won't be used.
+   */
+  handler?: LoggerInterface;
+}
+```
+
+#### 1. level
+
+This option sets the log level based on the [RFC 5424](https://datatracker.ietf.org/doc/html/rfc5424) syslog standard. It defines the severity of log messages that should be recorded. Available log levels are:
+
+- **emergency**: The highest severity level, indicating a system-wide emergency.
+- **alert**: Indicates an urgent condition that requires immediate attention.
+- **critical**: Highlights critical issues that need immediate action.
+- **error**: Used to log error messages that could affect functionality.
+- **warning**: Logs warnings about potential issues that should be monitored.
+- **notice**: Records significant but non-critical events.
+- **info**: Logs general informational messages indicating normal operation.
+- **debug**: The most detailed log level, useful for tracing and debugging the application.
+
+**Example:**
+Setting the log level to `error` ensures only error messages and above (i.e., critical, alert, emergency) will be captured:
+
+```json
+{
+  "logger": {
+    "level": "error"
```

Comment:
```suggestion
    "level": "error"
```
one last catch, wouldn't `verbosity` be more descriptive as we're talking specifically about log verbosity levels here/